src/pages/correspondent/Forms/CorrespondentForm.js

```diff
@@ -191,6 +193,26 @@ const CorrespondentForm = ({ labels, editMode, maxAccess, setEditMode, setStore,
                 error={formik.touched.currencyId && Boolean(formik.errors.currencyId)}
               />
             </Grid>
+            <Grid item xs={12}>
+              <ResourceComboBox
+                endpointId={RemittanceSettingsRepository.CorrespondentGroup.qry}
+                name='cgId'
+                label={labels.group}
+                valueField='recordId'
+                displayField={['reference', 'name']}
+                columnsInDropDown={[
+                  { key: 'reference', value: 'Reference' },
+                  { key: 'name', value: 'Name' }
+                ]}
+                values={formik.values}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('cgId', newValue?.recordId)
+                  formik.setFieldValue('cgName', newValue?.cgName)
```

Comment:
no need for cgName and cgRef in the comboboxes, ids are enough for the selection